starknet_py/utils/typed_data.py

```diff
@@ -354,17 +457,67 @@ def _get_merkle_tree_leaves_type(self, context: TypeContext) -> str:
                 f"Key {key} is not defined in type {parent} or multiple definitions are present."
             )
 
-        if target_type.contains is None:
-            raise ValueError("Missing 'contains' field in target type.")
+        if not isinstance(target_type, (EnumParameter, MerkleTreeParameter)):
+            raise ValueError("Target type is not an enum or merkletree type.")
+        return target_type
 
-        return target_type.contains
+    def _encode_enum(self, value: dict, context: TypeContext):
+        if len(value.keys()) != 1:
+            raise ValueError(
+                f"'{BasicType.ENUM.name}' value must contain a single variant."
+            )
+
+        variant_name, variant_data = next(iter(value.items()))
+        variants = self._get_enum_variants(context)
+        variant_type = next(
```

Comment:
nit:
```suggestion
        variant_definition = next(
```
Or something similar, "type" is a bit misleading here.